api/v1beta1/clusterclass_types.go

```diff
@@ -270,8 +270,16 @@ type MachineHealthCheckClass struct {
 	// +kubebuilder:validation:Pattern=^\[[0-9]+-[0-9]+\]$
 	UnhealthyRange *string `json:"unhealthyRange,omitempty"`
 
-	// Machines older than this duration without a node will be considered to have
-	// failed and will be remediated.
+	// NodeStartupTimeout allows to set the maximum time for MachineHealthCheck
+	// to consider a Machine unhealthy if a corresponding Node isn't associated
+	// through a `Spec.ProviderID` field.
+	//
+	// The duration set in this field is compared to the greatest of:
+	// - Cluster's infrastructure and control plane ready condition timestamp (if and when available)
```

Comment:
This is not true. We use the ControlPlaneInitializedCondition instead of the ControlPlaneReadyCondition

Can you please open a follow-up PR to correct the doc?

Reply:
Thx for extending the doc!